python/HeaderService/hslib_salobj.py

```diff
@@ -475,7 +475,23 @@ def collect_from_HeaderService(self):
             self.metadata['RAEND'] = ra
             self.metadata['DECEND'] = dec
         else:
-            self.log.info("No 'ELEND' and 'AZEND'")
+            self.log.info("No 'ELEND' and 'AZEND' to compute RA/DEC END")
+
+        if 'RA' and 'DEC' and 'ROTPA' in self.metadata:
```

Comment:
I don't think this is doing what you expect. I think it's only checking that `ROTPA` is in `self.metadata`. The `RA` and `DEC` are `True` independently. I think you either need three separate tests or you need to extract the keys from self.metadata as a set and ask if the `{"RA", "DEC", "ROTPA"}` set is in that set.  This construct is used elsewhere.

Reply:
Thanks for catching that. It's been fixed.